Added Like/Dislike Feature for recipes

beckystar4 opened the pull request:
Uses same functionality as inventory. Created model for recipe library. Saves to user's liked recipes

## Trello Card

[Replace this text with a link to your trello card(s).](https://trello.com/c/G4DJaAkn/12-create-drink-rating-function-like-dislike)
https://trello.com/c/UA1HXWKr/14-save-users-liked-recipes

## Description

Added recipe library model. 
Uses same functionality as inventory (ingredient items). Save's user likes.

beckystar4 committed:
Added Like/Dislike Feature

Uses same functionality as inventory. Created model for recipe library.
Saves to user's liked recipes

beckystar4 committed:
Updated navbar

Added functionality to see the liked recipes/ingredients if logged in

JMGbaum commented on config/routes.rb:
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  get 'librarys/new'
```
```suggestion
```

JMGbaum commented on config/routes.rb:
```diff
@@ -16,6 +17,7 @@
 
   # post  "/inventory_item", to: "ingredient_items#create"
   resources :inventory_items, only: %i[create], controller: :ingredient_items
+  resources :librarys, only: %i[create], controller: :librarys
```
```suggestion
  resources :librarys, only: %i[create new], controller: :librarys
```

JMGbaum commented on db/migrate/20230325175450_create_libraries.rb:
We may want to add a field for a rating, whatever form we want that to take

JMGbaum commented on app/controllers/librarys_controller.rb:
Is this not supposed to be `Libraries` instead of `Librarys`? This goes for the other files too.

beckystar4 replied:
I accidentally spelt it wrong lol! I meant to call it RecipeLibrary but then I messed something up and I don't know how easy it is to refactor

JMGbaum replied:
I think in the current state there might be errors since the table in the migration is called `libraries` but you are using `librarys` everywhere else. Rails is capable of pluralizing irregular words ([see here](https://guides.rubyonrails.org/active_record_basics.html#naming-conventions)). Fixing this should just be a matter of updating the filenames and any calls to the model/controller in the code, but I'd rather address it now before we build upon it.

I'd also suggest rethinking the name a little bit and changing it to something like `SavedRecipe` or `UserRecipe`, since using `Library` as the model name implies that user will have multiple libraries rather than multiple saved recipes. If you'd like, I can help you fix your migration and naming conventions tomorrow either before or after our group meeting.

beckystar4 committed:
Update config/routes.rb

Co-authored-by: Jordan <32182454+JMGbaum@users.noreply.github.com>

beckystar4 committed:
Update config/routes.rb

Co-authored-by: Jordan <32182454+JMGbaum@users.noreply.github.com>

beckystar4 committed:
Update app/views/layouts/_navbar.html.erb

Co-authored-by: Jordan <32182454+JMGbaum@users.noreply.github.com>

beckystar4 committed:
Merge pull request #16 from JMGbaum/updated_usermodel

Updated nav bar. Goes into the like/dislike branch

JMGbaum commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -18,19 +18,30 @@
                     </li>
 
-                    <%# TODO: Add link %>
+                    <% if logged_in? %>
```
Where is this `logged_in?` method defined?

JMGbaum replied:
Nvm, I just saw this was answered in #16.

beckystar4 committed:
Refactored Name/Delete Button

This has the refactored name for the user recipes and a delete from library button

beckystar4 committed:
Merge branch 'develop-2' into like/dislike-recipes

beckystar4 committed:
Update index.html.erb

beckystar4 committed:
delete unused files
